tests/services/apiService.test.js

```diff
@@ -40,3 +40,113 @@ describe("fetchQuestCategoryData function", () => {
     expect(result).toEqual(mockResponse);
   });
 });
+
+describe("fetchLeaderboardRankings function", () => {
+  beforeEach(() => {
+    fetch.mockClear();
+  });
+
+  it("should fetch and return data for valid parameters", async () => {
+    // Mock data creation for a successful response
+    const mockData = {
+      rankings: [
+        { address: '0x123abc', xp: 100, achievements: 5 },
+        { address: '0x456def', xp: 80, achievements: 3 },
+        { address: '0x789ghi', xp: 120, achievements: 7 },
+      ],
+      first_elt_position: 1,
+    };
+
+    fetch.mockResolvedValueOnce({
+      json: () => Promise.resolve(mockData),
+    });
+
+    const params = { addr: '', page_size: 10, shift: 0, duration: 'week' };
+    const result = await fetchLeaderboardRankings(params);
+
+    expect(fetch).toHaveBeenCalledWith(
+      `${API_URL}/leaderboard/get_ranking?addr=&page_size=10&shift=0&duration=week`
+    );
+    expect(result).toEqual(mockData);
+  });
+
+  it("should handle API returning no response", async () => {
+    // Mock fetch response with no data
+    fetch.mockResolvedValueOnce(undefined);
+
+    const params = { addr: 'invalidAddr', page_size: 10, shift: 0, duration: 'week' };
+    const result = await fetchLeaderboardRankings(params);
+
+    expect(fetch).toHaveBeenCalledWith(
+      `${API_URL}/leaderboard/get_ranking?addr=invalidAddr&page_size=10&shift=0&duration=week`
+    );
+    expect(result).toBeUndefined();
+  });
+
+  it("should handle API returning response in unexpected format", async () => {
```

Comment:
what is unexpected in this format?

Reply:
you can try setting the following params invalid one by one to test this 
- send page size as a negative number
- duration as a random string and not - 'all' | 'week' | 'month'

Reply:
Okay added the requested changes kindly review